examples/thread_test/src/main.rs

```diff
@@ -0,0 +1,230 @@
+#![feature(thread_local)]
+
+use std::cell::{Cell, RefCell};
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::thread;
+use std::thread::available_parallelism;
+use std::time::Duration;
+
+#[cfg(target_os = "hermit")]
+use hermit as _;
+use log::info;
+
+// Inspired by the Rust standard TLS tests:
+// https://github.com/rust-lang/rust/tree/master/library/std/tests/thread_local
+
+// --- Main TLS demonstration ---
+
+// TLS variables of various types.
+#[thread_local]
+static TLS_VALUE: Cell<i32> = Cell::new(0);
+#[thread_local]
+static TLS_U64: Cell<u64> = Cell::new(0);
+#[thread_local]
+static TLS_F64: Cell<f64> = Cell::new(0.0);
+#[thread_local]
+static TLS_BOOL: Cell<bool> = Cell::new(false);
+#[thread_local]
+static TLS_CHAR: Cell<char> = Cell::new('A');
+#[thread_local]
+static TLS_STRING: Cell<&'static str> = Cell::new("Initial String");
+#[thread_local]
+static TLS_U8: Cell<u8> = Cell::new(0);
+#[thread_local]
+static TLS_U64_2: Cell<u64> = Cell::new(0xdeadbeef);
```

Comment:
Can we use the [`thread_local!`](https://doc.rust-lang.org/stable/std/macro.thread_local.html) instead and remove the `thread_local` night feature?

```suggestion
thread_local! {
    static TLS_VALUE: Cell<i32> = Cell::new(0);
    static TLS_U64: Cell<u64> = Cell::new(0);
    static TLS_F64: Cell<f64> = Cell::new(0.0);
    static TLS_BOOL: Cell<bool> = Cell::new(false);
    static TLS_CHAR: Cell<char> = Cell::new('A');
    static TLS_STRING: Cell<&'static str> = Cell::new("Initial String");
    static TLS_U8: Cell<u8> = Cell::new(0);
    static TLS_U64_2: Cell<u64> = Cell::new(0xdeadbeef);
}
```